src/scanner/utility.ts

```diff
@@ -0,0 +1,1031 @@
+import { trackEventData, trackEventDataSet } from "./dbOperations";
+import { bnToBn, extractTime } from "@polkadot/util";
+import {accuracyFormat} from "../scanner/formatBalance";
+
+let blockTime;
+
+// Convert a block number to date
+async function convertBlockToDate(api, blockNumber, date) {
+	blockTime = blockTime ? blockTime : await api.consts.babe.expectedBlockTime;
+	const value = blockTime.mul(bnToBn(blockNumber)).toNumber();
+	const time = extractTime(Math.abs(value));
+	const { days } = time;
+	const d = new Date(date);
+	d.setDate(d.getDate() + days);
+	return d;
+}
+
+// store in db all the relevant data for mint unique nft
+export async function trackUniqueMintData(
+	eventData,
+	api,
+	params,
+	date,
+	owner,
+	txHash,
+	blockNumber
+) {
+	try {
+		const tokenId = JSON.stringify(eventData[1]); // tokenId in format [17,5,0] - [collectionId, seriesId, serialNo]
+		const imgUrl = api.registry
+			.createType(params[3].type, params[3].value)
+			.toHuman();
+		const tokenData = {
+			eventData: {
+				imgUrl: imgUrl,
+				date: date,
+				owner: owner,
+				txHash: txHash,
+			},
+			eventType: "NFT_CREATED",
+		};
+		const type = 0;
+		await trackEventData(
+			tokenId,
+			type,
+			blockNumber,
+			JSON.stringify(tokenData),
+			owner
+		);
+	} catch (e) {
+		console.log(
```

Comment:
I think we should re throw the error as this point, so the nftsModel can catch it all and handle them accordingly